jax/_src/scipy/stats/_rankdata.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2023 The JAX Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from functools import partial
+from typing import Optional
+
+import scipy
+from jax import numpy as jnp, jit
+from jax._src import dtypes
+from jax._src.numpy.lax_numpy import _check_arraylike
+from jax._src.numpy.util import _wraps
+from jax._src.typing import Array
+
+def invert_permutation(i: Array) -> Array:
+    """Helper function that inverts a permutation array."""
+    return jnp.empty_like(i).at[i].set(jnp.arange(i.size, dtype=i.dtype))
```

Comment:
Here and throughout, please use two-space indentations for consistency with the rest of the package.